### PR TITLE
docs(forge): improve `ensure_git_status_clean` error message

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

#4223

> I feel like this is in the top 5 questions asked in the telegram chat, so tried to add even more information to the error message

### CODE REVIEW

File: cli/src/cmd/forge/install.rs [hunk 1]
```diff
@@ -305,7 +305,18 @@ fn commit_after_install(libs: &Path, target_dir: &str, tag: Option<&str>) -> eyr
 
 pub fn ensure_git_status_clean(root: impl AsRef<Path>) -> eyre::Result<()> {
     if !git_status_clean(root)? {
-        eyre::bail!("This command requires clean working and staging areas, including no untracked files. Modify .gitignore and/or add/commit first, or add the --no-commit option.")
+        eyre::bail!(
+            "\
+This command's target directory is, or is in an already initialized git repository,
```
Comment (rkrasiuk) [inline cli/src/cmd/forge/install.rs:310]:
```suggestion
The target directory is a part of or on its own an already initialized git repository,
```

File: cli/tests/fixtures/can_detect_dirty_git_status_on_init.stderr [hunk 2]
```diff
@@ -1,2 +1,10 @@
 Error: 
-This command requires clean working and staging areas, including no untracked files. Modify .gitignore and/or add/commit first, or add the --no-commit option.
+This command's target directory is, or is in an already initialized git repository,
```
Comment (rkrasiuk) [inline cli/tests/fixtures/can_detect_dirty_git_status_on_init.stderr:2]:
```suggestion
The target directory is a part of or on its own an already initialized git repository,
```

**Review by rkrasiuk — COMMENTED** [review]
smol nit

**Review by rkrasiuk — APPROVED** [review]

**Review by mattsse — APPROVED** [review]
I thought about this a bit.
Even if verbose this is very clear and helpful and an improvement

**Pull request merged**